spring-cloud-stream-binders/spring-cloud-stream-binder-redis/src/main/java/org/springframework/cloud/stream/binder/redis/RedisMessageChannelBinder.java

```diff
@@ -359,82 +286,54 @@ private void doRegisterProducer(final String name, MessageChannel moduleOutputCh
-			Properties properties) {
-		if (logger.isInfoEnabled()) {
-			logger.info("binding replier: " + name);
-		}
-		validateConsumerProperties(name, properties, SUPPORTED_REPLYING_CONSUMER_PROPERTIES);
-		RedisPropertiesAccessor accessor = new RedisPropertiesAccessor(properties);
-		MessageProducerSupport adapter = createInboundAdapter(accessor, "queue." + applyRequests(name));
-		this.doRegisterConsumer(name, name, requests, adapter, accessor);
-
-		RedisQueueOutboundChannelAdapter replyQueue = new RedisQueueOutboundChannelAdapter(
-				RedisMessageChannelBinder.parser.parseExpression("headers['" + BinderHeaders.REPLY_TO + "']"),
-				this.connectionFactory);
-		replyQueue.setBeanFactory(this.getBeanFactory());
-		replyQueue.setIntegrationEvaluationContext(this.evaluationContext);
-		replyQueue.afterPropertiesSet();
-		this.doRegisterProducer(name, replies, replyQueue, accessor);
-	}
-
-	@Override
 	public void destroy() {
 		stopBindings();
 	}
 
 	private class SendingHandler extends AbstractMessageHandler {
 
-		private final MessageHandler delegate;
-
-		private final String replyTo;
+		private final String bindingName;
 
 		private final PartitioningMetadata partitioningMetadata;
 
+		private final RedisPropertiesAccessor accessor;
+
+		private final Map<String, RedisQueueOutboundChannelAdapter> adapters = new HashMap<>();
 
-		private SendingHandler(MessageHandler delegate, String replyTo, RedisPropertiesAccessor properties) {
-			this.delegate = delegate;
-			this.replyTo = replyTo;
+		private SendingHandler(String bindingName, RedisPropertiesAccessor properties) {
+			this.bindingName = bindingName;
+			this.accessor = properties;
 			this.partitioningMetadata = new PartitioningMetadata(properties, properties.getNextModuleCount());
 			this.setBeanFactory(RedisMessageChannelBinder.this.getBeanFactory());
+			refreshChannelAdapters();
 		}
 
 		@Override
 		protected void handleMessageInternal(Message<?> message) throws Exception {
 			MessageValues transformed = serializePayloadIfNecessary(message);
 
-			if (replyTo != null) {
-				transformed.put(BinderHeaders.REPLY_TO, this.replyTo);
-			}
 			if (this.partitioningMetadata.isPartitionedModule()) {
-
 				transformed.put(PARTITION_HEADER, determinePartition(message, this.partitioningMetadata));
 			}
 
 			byte[] messageToSend = embeddedHeadersMessageConverter.embedHeaders(transformed,
 					RedisMessageChannelBinder.this.headersToMap);
-			delegate.handleMessage(MessageBuilder.withPayload(messageToSend).copyHeaders(transformed).build());
+			
+			refreshChannelAdapters();
+			for (RedisQueueOutboundChannelAdapter adapter : adapters.values()) {
+				adapter.handleMessage((MessageBuilder.withPayload(messageToSend).copyHeaders(transformed).build()));
+			}
 		}
 
+		private void refreshChannelAdapters() {
+			Set<String> groups = redisOperations.boundSetOps(CONSUMER_GROUPS_KEY_PREFIX + bindingName).members();
```

Comment:
If groups (and their subscriptions) are not durable - based on https://github.com/spring-cloud/spring-cloud-stream/pull/276/files#diff-abf0f621ebbf51a773c2d6ab5129ac24R248, then maybe it makes sense to shut down adapters for groups that aren't listening anymore ?